saleor/order/utils.py

```diff
@@ -65,12 +67,51 @@ def decorator(*args, **kwargs):
                 discount = get_voucher_discount_for_order(order)
             except NotApplicable:
                 discount = zero_money(order.currency)
-            order.discount = discount
-        return func(*args, **kwargs)
+        return func(*args, **kwargs, discount=discount)
 
     return decorator
 
 
+def get_voucher_discount_assigned_to_order(order: Order):
```

Comment:
Again, I would consider if we can move all purely discount-related functions to `saleor/order/utils/discounts.py`.